project/src/main/java/com/google/autograder/servlets/GetAssignmentDetailsServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.autograder.servlets;
+
+import java.net.URL;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.autograder.servlets.helpers.Services;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+@WebServlet("/getAssignmentDetails")
+public final class GetAssignmentDetailsServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestURL = (String) request.getHeader("Referer");
```

Comment:
Similarly the course ID and assignment ID should be passed as query parameters, rather than extracted from the referer (and the same for the rest of the servlets)